corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -214,27 +217,38 @@ def get_userdata_autoselect(key, session_id, mode):
         ]
         return datum, assertions
 
+    @memoized
+    def include_post_in_entry(self, module_id):
+        module = self.app.get_module_by_unique_id(module_id)
+        loads_registry_case = module_loads_registry_case(module)
+        using_inline_search = module_uses_inline_search(module)
+        sync_on_form_entry = (
+            module_offers_search(module)
+            and case_search_sync_cases_on_form_entry_enabled_for_domain(self.app.domain)
+        )
+        return (using_inline_search or sync_on_form_entry) and not loads_registry_case
+
     def entry_for_module(self, module):
         # avoid circular dependency
         from corehq.apps.app_manager.models import Module, AdvancedModule
         results = []
-        loads_registry_case = module_loads_registry_case(module)
-        using_inline_search = module_uses_inline_search(module) and not loads_registry_case
         for form in module.get_suite_forms():
             e = Entry()
             e.form = form.xmlns
 
             if module.report_context_tile:
                 from corehq.apps.app_manager.suite_xml.features.mobile_ucr import get_report_context_tile_datum
                 e.datums.append(get_report_context_tile_datum())
-
-            if form.requires_case() and using_inline_search:
+            if form.requires_case() and self.include_post_in_entry(module.get_or_create_unique_id()):
+                case_session_var = self.get_case_session_var_for_form(form)
                 from corehq.apps.app_manager.suite_xml.post_process.remote_requests import (
-                    RemoteRequestFactory, RESULTS_INSTANCE_INLINE
+                    RemoteRequestFactory, RESULTS_INSTANCE_INLINE, RESULTS_INSTANCE
                 )
-                case_session_var = self.get_case_session_var_for_form(form)
+                storage_instance = RESULTS_INSTANCE_INLINE if module_uses_inline_search(module) \
+                    else RESULTS_INSTANCE
```

Comment:
@stephherbers what made you chose 'RESULTS_INSTANCE' here?

Reply:
I chose that because it was the default for `RemoteRequestFactory`, but is there one it should be instead?

Reply:
In the context of an `entry` the `results` instance doesn't exist so this should `casedb`. Based on our recent analysis, this is only used for inline search but it's probably worth making it 'casedb' anyway for the sake of futureproofing.